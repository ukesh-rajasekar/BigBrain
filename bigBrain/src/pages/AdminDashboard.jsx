import React, { useEffect, useState } from 'react';
import { Container, Row } from 'react-bootstrap';
import Button from '../components/button';
import GameCards from '../components/game/gameCard';
import Input from '../components/Input';
import Navbar from '../components/navBar';
import { oncreate } from '../services/Admin/games';
import {
  fetchAllGames,
  fetchAllGamesByIds,
} from '../services/games/gameService';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome'
import { faPlusCircle } from '@fortawesome/free-solid-svg-icons'
function AdminDashboard (props) {
  const [games, setGames] = useState({});
  const [gameObj, setGameObj] = useState({ name: '' });

  const [newGame, setNewGame] = useState(false);

  const fetchGames = () => {
    fetchAllGames().then((data) => {
      if (!data) return
      const ids = [];
      for (const quiz of data?.quizzes) {
        ids.push(quiz.id);
      }
      if (ids.length === 0) return
      fetchAllGamesByIds(ids).then((values) => {
        const games = [];
        let count = 0;
        for (const value of values) {
          let gameDetails = value.value;
          gameDetails = { ...gameDetails, id: ids[count] };
          count = count + 1;
          games.push(gameDetails);
        }
        setGames(games);
      })
    })
  };
  const setStateValue = (item, value) => {
    console.log(item, value);
    setGameObj({ ...gameObj, [item]: value });
  };

  useEffect(() => {
    fetchGames()
    return () => {};
  }, [gameObj]);

  const deleteGame = (gameid) => {
    const newGames = games.map((game, idx) => {
      if (game.id !== gameid) { return game } else return null
    })
    console.log(newGames);
    setGames(newGames)
  }
  return (
    <React.Fragment>
      <Navbar></Navbar>
          <h1 className = 'dashboard-title' hidden>Dashboard</h1>
          <div>
              <div>
                <Button
                  name = 'createGame'
                  buttonText={!newGame ? <>Create Game <FontAwesomeIcon color="white" icon={faPlusCircle} /></> : ' Cancel '}
                  buttonAction={() => setNewGame(!newGame)}
                />
              </div>

        {newGame && (
          <Input
            name="name"
            placeholder="Enter the game`s name"
            className="new-game"
            type="text"
            handleChange={setStateValue}
          />
        )}
      </div>
      <div>
        {newGame && <Button name='confirmGame' buttonText="Submit" buttonAction={() => oncreate(gameObj).then(() => fetchGames())} />}
      </div>
      <Container><Row>
      {Object.entries(games).map(([idx, gameData]) => {
        return (
          gameData ? <GameCards key={gameData.id} gameData= {gameData} handleDeleteGame={deleteGame} /> : null
        )
      })}
          </Row></Container>
    </React.Fragment>
  );
}
export default AdminDashboard;
