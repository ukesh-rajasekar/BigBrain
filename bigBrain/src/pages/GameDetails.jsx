import React, { useEffect, useState } from 'react';
import { useHistory, useParams } from 'react-router';
import {
  deleteGameById,
  updateGameQuestions,
} from '../services/Admin/gamehelper';
import { getCopy } from '../services/helpers';
import GameQuestion from '../components/game/gameQuestions';
import Modal from 'react-modal';
import { fetchQuizData } from '../services/games/gameService';
import Button from '../components/button';
import Input from '../components/Input';
import PageNotFound from '../components/PageNotFound';
import { questionFormat } from '../constants/questionFormat';
import { v4 as uuidv4 } from 'uuid';
import Navbar from '../components/navBar';

function GameDetails () {
  const { gameId } = useParams();
  const [gameData, setGameData] = useState(null);
  const [name] = useState('');
  const history = useHistory();
  let subtitle;
  const [newQuestions, setnewQuestions] = useState([]);
  const [modalIsOpen, setIsOpen] = React.useState(false);
  const [newQuestion, setNewQuestion] = useState('');

  useEffect(() => {
    fetchQuizData(gameId).then((gameData) => {
      if (gameData) {
        setGameData(gameData);
        if (gameData?.questions?.length !== 0) {
          setnewQuestions(getCopy(gameData.questions));
        }
      }
    });
    return () => {};
  }, [gameId]);

  function openModal () {
    setIsOpen(true);
  }
  function afterOpenModal () {
    // references are now sync'd and can be accessed.
    subtitle.style.color = '#f00';
  }
  function closeModal () {
    setIsOpen(false);
  }
  const handleNewQuestionChange = (item, value) => {
    setNewQuestion(value);
  };
  const addQuestion = () => {
    const newQues = getCopy(questionFormat);
    newQues.id = uuidv4();
    newQues.question.value = newQuestion;

    updateGameQuestions(gameId, {
      questions: [...newQuestions, newQues],
      name: name,
      thumbnail: '',
    }).then((data) => {
      if (JSON.stringify(data) === '{}') {
        setnewQuestions([...newQuestions, newQues]);
      }
    });
  };

  const deleteQuestionOfIdx = (idx) => {
    const temp = getCopy(newQuestions);
    temp.splice(idx, 1);
    updateGameQuestions(gameId, {
      questions: temp,
      name: name,
      thumbnail: '',
    }).then((data) => {
      if (JSON.stringify(data) === '{}') {
        setnewQuestions(getCopy(temp));
      }
    });
  };

  const deleteGame = () => {
    deleteGameById(gameId).then((data) => {
      if (JSON.stringify(data) === '{}') {
        history.push('/admin');
      }
    });
  };
  if (!gameData) return <PageNotFound />;
  return (
    <div className="gameWrapper">
            <Navbar></Navbar>

      <div className="gameContainer">
        <h3>{gameData.name}</h3>
        {newQuestions.map((question, idx) => {
          return (
              <div key={ idx}>
              <GameQuestion
                key={idx}
                question={question}
                quesId={question.id}
                index={idx}
                handleDelete={deleteQuestionOfIdx}
                  />

            </div>
          );
        })}
        <Button name="createQuestion" buttonText="Create question" buttonAction={openModal}></Button>
        <Button name="deleteGame" variant="danger" buttonText="Delete game" buttonAction={deleteGame}></Button>
        <Modal
          isOpen={modalIsOpen}
          onAfterOpen={afterOpenModal}
          onRequestClose={closeModal}
          style={{
            content: {
              top: '50%',
              left: '50%',
              right: 'auto',
              bottom: 'auto',
              marginRight: '-50%',
              transform: 'translate(-50%, -50%)',
            },
          }}
          contentLabel="Example Modal"
        >
          <h2 ref={(_subtitle) => (subtitle = _subtitle)}>Add Your Question</h2>
          <form>
            <Input
              name="question"
              placeholder="Enter the question"
              className="addQuestionQuestion"
              type="text"
              handleChange={handleNewQuestionChange}
            />
          </form>
          <Button name="addQuestion" buttonText="Add Question" buttonAction={addQuestion}></Button>
          <Button name="closeQuestion" buttonText="close" buttonAction={closeModal}></Button>
        </Modal>
      </div>
    </div>
  );
}

export default GameDetails;
