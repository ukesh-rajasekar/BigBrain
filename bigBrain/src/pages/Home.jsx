import React, { useState, useEffect } from "react";
import Button from "../components/button";
import Input from "../components/Input";
import Navbar from "../components/navBar";
import Popup from "../components/popups";
import { urls } from "../constants/urls";
import { doPost, doGet } from "../services/apiRequests";
import { showToast } from "../services/toastServices";
import { useHistory } from "react-router";
import {
  fetchAllGames,
  fetchAllGamesByIds,
} from "../services/games/gameService";
import { Link } from "react-router-dom";

// import Navbar from '../Components/navbar'

export const Home = () => {
  const history = useHistory();
  const [games, setGames] = useState({});
  const [quizId, setQuizId] = useState(null);
  const [sessionStatus, setSessionStatus] = useState(false);
  const [sessionId, setSessionId] = useState("");
  const [sessionOpen, setSessionOpen] = useState(false);
  const [sessionClose, setSessionClose] = useState(false);
  
const fetchGames = () => {
      fetchAllGames().then((data) => {
        if(!data) return
      const ids = [];
      for (const quiz of data?.quizzes) {
        ids.push(quiz.id);
        }
        if (ids.length === 0) return
      fetchAllGamesByIds(ids).then((values) => {
        const games = [];
        let count = 0;
        for (const value of values) {
          let gameDetails = value.value;
          gameDetails = { ...gameDetails, id: ids[count] };
          count = count + 1;
          games.push(gameDetails);
        }
        setGames(games);
      })
    })
  };

  useEffect(() => {
      fetchGames()
    return () => {};
  }, []);




  const onstart = (gameId) => {
    setQuizId(gameId)
    doPost(urls.gameSession + `/${gameId}/start`).then((res) => {
      if (res.status === 200) {
        console.log("created");
        showToast(`Game started`, "success");
        doGet(urls.gameSession + `/${gameId}`).then((res) => {
          if (res.status === 200) {
            res.json().then((data) => {
              setSessionId(data.active);
              setSessionOpen(true);
            });
          }
        });
        setSessionStatus(!sessionStatus);
      } else {
        //console.log('Invalid request')
        showToast("Game cannot be created", "error");
      }
    });
  };
  const onadvance = (gameId) => {
    doPost(`${urls.gameSession}/${gameId}/advance`).then((res)=> {
      if (res.status === 200) {
        console.log("Advancing");
        showToast(`Game Advanced`, "info");
      } else {
        showToast('Game Ended', 'info')
        setSessionStatus(!sessionStatus);
        setSessionOpen(false);
        setSessionClose(true);
      }
    })
  }
  const onend = (gameId) => {
    doPost(urls.gameSession + `/${gameId}/end`).then((res) => {
      if (res.status === 200) {
        console.log("ended");
        showToast(`Game ended`, "success");
        // doPost(urls.gameSession + '997161454').then((res) => {
        //     if (res.status === 200) {
        //         res.json().then((data) => {
        //             alert(data.active)
        //         })
        //     }
        // })
        setSessionStatus(!sessionStatus);
        setSessionOpen(false);
        setSessionClose(true);
      } else {
        //console.log('Invalid request')
        showToast("You have already ended the game", "error");
      }
    });
  };

  const gotoresults = (sessionId, quizId) => {
    setSessionClose(false);
    console.log(sessionId);
    history.push(`/admin/${quizId}/${sessionId}/results`);
  };

  return (
    <React.Fragment>
      <Navbar />
      <h1>Home</h1>
      <ul>
        <li>
          <Link to="/login">Login</Link>
        </li>
        <li>
          <Link to="/register">Register</Link>
        </li>
      </ul>
     
      <div>
        <div className="gamesWrapper">
          <div className="gamesContainer">
            {Object.entries(games).map((value) => {
              return <React.Fragment key={value[1].id}><h3 key={value[0]}>{value[1].name}</h3>
              <div>{(<Button buttonText = 'Start game' buttonAction = {()=> onstart(value[1].id)} />)}</div>
              
        <div>{(<Button buttonText = 'End game' buttonAction = {()=> onend (value[1].id)} />)}</div>
              </React.Fragment>
            }) }
          </div>
        </div>

        <div>
          {sessionOpen && (
            <Popup
              content={
                <>
                  <b>Session {sessionId} Started!!!</b>
                  <Button
                    buttonText="Copy session link"
                    buttonAction={() =>
                      navigator.clipboard.writeText(sessionId)
                    }
                  />
                  <div>{(<Button buttonText = 'Advance game' buttonAction = {()=> onadvance(quizId)} />)}</div>
                </>
              }
              handleClose={() => setSessionOpen(false)}
            />
          )}
        </div>

        <div>
          {sessionClose && (
            <Popup
              content={
                <>
                  <b>Would you like to view the results?</b>
                  <Button
                    buttonText="Yes"
                    buttonAction={() => {
                      gotoresults(sessionId, quizId);
                    }}
                  />
                  <Button
                    buttonText="No"
                    buttonAction={() => {
                      setSessionClose(false);
                    }}
                  />
                </>
              }
              handleClose={() => setSessionClose(false)}
            />
          )}
        </div>
      </div>
    </React.Fragment>
  );
};
