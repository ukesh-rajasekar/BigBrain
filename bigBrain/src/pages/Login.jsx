import React, { useContext, useState } from 'react'

import { useHistory } from 'react-router';
import Input from '../components/Input'
import Button from '../components/button'
import { Authenticator } from '../contexts/Auth';
import { loginAdmin } from '../services/Auth/authServices';
import './styles/login.css';
import { Card, Row, Container, Col } from 'react-bootstrap'
import NavbarPlayers from '../components/navBarPlayers';

export const validateInput = (str = '') => {
  return str.includes('@');
}

export const Login = () => {
  const history = useHistory();
  const auth = useContext(Authenticator);
  const [formValues, setForm] = useState({ email: '', password: '' })
  const setStateValue = (item, value) => {
    setForm({ ...formValues, [item]: value })
  }

  const onsubmit = () => {
    loginAdmin(formValues).then((data) => {
      if (data?.token) {
        auth.signIn(data);
        history.push('/admin')
      } else {
        alert(data.error)
      }
    });
  }

  return (
    <Container fluid="md">
        <NavbarPlayers></NavbarPlayers>

      <Row className="justify-content-md-center">
        <Col>
      <Card className=' text-center'
        bg={'light'}>
        <Card.Body className='container'>
          <Card.Header>Hello BigBrains, please log in</Card.Header>
            <Input
              name='email'
              placeholder='Email'
              className='email'
              type='text'
              handleChange={setStateValue}
            />
            {formValues.email && !validateInput(formValues.email) ? <p>Invalid email !</p> : null}
            <Input
              name='password'
              placeholder='Password'
              className='password'
              type='password'
              handleChange={setStateValue}
            />
            <Button name='login' buttonText='Log In' buttonAction={onsubmit} />
            <Button name='signup' buttonText='Sign Up' buttonAction={() => history.push('/register')} />
          </Card.Body>
          </Card >
          </Col>
     </Row>
</Container>
  );
};
