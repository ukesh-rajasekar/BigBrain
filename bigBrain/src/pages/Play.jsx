import React, {useState} from 'react'
import { useHistory, useRouteMatch } from 'react-router-dom'
import Button from '../components/button'
import Input from '../components/Input'

function Play () {
    const [playerName, setPlayerName] = useState({ name: 'Guest'})
    const [sessionId, setSessionId] = useState('')

    const history = useHistory()
    const {url}= useRouteMatch()
      const setStateValue = (item, value) => {
        setPlayerName({ ...playerName, [item]: value })
    }

    const setSessionValue = (item,value) => {
        setSessionId(value)
    }
   
  return (
    <React.Fragment>
      <div className='join-wrapper'>
          <div className='join-container'>
            <p className='header'>Join Session</p>
            <Input
              name='sessionId'
              placeholder='Session Id'
              className='sessionId'
              type='text'
              handleChange={ setSessionValue}
            />
            <Input
              name='name'
              placeholder='Name'
              className='name'
              type='text'
              handleChange={setStateValue}
            />
            <Button buttonText='Join game' buttonAction={()=>history.push(`${url}/${playerName.name}/${sessionId}`)} />
          </div>
        </div>
    </React.Fragment>
  )
}

export default Play;