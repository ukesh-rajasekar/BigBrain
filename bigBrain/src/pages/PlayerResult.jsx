import React, { useEffect, useState } from 'react'
import { getCopy } from '../services/helpers';
import { getPlayerResults } from '../services/Player/playerServices';
import PropTypes from 'prop-types'
import NavbarPlayers from '../components/navBarPlayers';

export default function PlayerResult ({ playerId, playerName, points }) {
  const QuestionPoints = getCopy(points)
  QuestionPoints.shift()
  const [results, setResults] = useState([])
  const [playerPoints, setPlayerPoints] = useState(0)

  useEffect(() => {
    getPlayerResults(playerId).then((data) => {
      setResults(data)
    })
    return () => {

    }
  }, [])

  useEffect(() => {
    caluculatePoints()
    return () => {

    }
  }, [results])

  const caluculatePoints = () => {
    let playerPoints = 0
    results.map((result, idx) => {
      if (result.correct) {
        playerPoints = playerPoints + QuestionPoints[idx]
      }
      console.log(playerPoints);
      setPlayerPoints(playerPoints);
      return null
    })
  }
  return (
    <div>
        <NavbarPlayers></NavbarPlayers>

            <h3>{playerName} Score</h3>
           {playerPoints} Points
           {results.map((result, idx) => {
             return <div key={idx}>
                Question {idx + 1}: {result.correct ? 'Correct' : 'Incorrect'}, Points: {result.correct ? QuestionPoints[idx] : 0}
               </div>
           })}
        </div>
  )
}

PlayerResult.propTypes = {
  playerId: PropTypes.any,
  playerName: PropTypes.any,
  points: PropTypes.any
}
