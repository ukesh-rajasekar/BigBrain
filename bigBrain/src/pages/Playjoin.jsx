import React, {useState} from 'react'
import Button from '../components/button'
import Input from '../components/Input'






function Playjoin () {
    const [playerName, setPlayerName] = useState({ name: ''})
    const [sessionId, setSessionId] = useState('')
      const setStateValue = (item, value) => {
        setPlayerName({ ...playerName, [item]: value })
    }

    const setSessionValue = (value) => {
        setPlayerName(value)
    }
   
  return (
    <React.Fragment>
      <div className='join-wrapper'>
          <div className='join-container'>
            <p className='header'>Join Session</p>
            <Input
              name='sessionId'
              placeholder='Session Id'
              className='sessionId'
              type='text'
              handleChange={ setSessionValue}
            />
            <Input
              name='name'
              placeholder='Name'
              className='name'
              type='text'
              handleChange={setStateValue}
            />
            <Button buttonText='Join game' />
          </div>
        </div>
    </React.Fragment>
  )
}

export default Playjoin;