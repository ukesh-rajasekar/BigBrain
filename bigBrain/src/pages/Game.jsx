import React, { useState, useEffect } from 'react';
import { Col, Container, Row } from 'react-bootstrap';
import { useParams } from 'react-router-dom';
import CurrentQuestion from '../components/player/CurrentQuestion';
import { getCopy } from '../services/helpers';
import {
  getPlayerId,
  pollGameStart,
  pollGameEnd,
} from '../services/Player/playerServices';
import { showToast } from '../services/toastServices';
import PlayerResult from './PlayerResult';

export default function Game () {
  const { playerName, sessionId } = useParams('Guest');
  const [playerId, setPlayerId] = useState(null);
  const [gameStarted, setGameStarted] = useState(false);
  const [questionPoints, setquestionPoints] = useState([]);
  const [questionPoint, setquestionPoint] = useState();
  const [gameEnded, setGameEnded] = useState(false);
  const points = (point) => {
    setquestionPoint(Number(point));
  };
  useEffect(() => {
    const temp = getCopy(questionPoints);
    temp.push(questionPoint);
    setquestionPoints(temp);
    console.log(temp);
    console.log(questionPoints);
    return () => {};
  }, [questionPoint]);
  useEffect(() => {
    getPlayerId(sessionId, playerName).then((data) => {
      if (data?.error) {
        showToast(data.error, 'error');
      } else {
        setPlayerId(data.playerId);

        showToast(`${playerName} Joined the game`, 'info');
        pollGameStart(data.playerId)
          .then(() => {
            showToast('Game Started', 'info');
            setGameStarted(true);
            pollGameEnd(data.playerId).then(() => {
              showToast('Game Ended', 'info');
              setGameEnded(true);
              // history.push(`${url}/result`)
            });
          })
          .catch(() => {
            showToast('Game Ended', 'info');
            setGameEnded(true);
            setGameStarted(true);
          });
      }
    });
    return () => {};
  }, []);
  if (!gameStarted) {
    return (
      <Container>
        <Row>
          <Col>
            <h3>Welcome {playerName}</h3>
            <h4>Waiting for the game to start</h4>
          </Col>
        </Row>
      </Container>
    );
  } else if (gameEnded) {
    return (
      <PlayerResult
        playerId={playerId}
        playerName={playerName}
        points={questionPoints}
      />
    );
  }
  return (
      <CurrentQuestion playerId={playerId} points={points} />
  );
}
