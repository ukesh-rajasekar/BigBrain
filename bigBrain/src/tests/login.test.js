import React from 'react'
import { fireEvent, render } from '@testing-library/react';
import { configure } from 'enzyme';
import {Login, validateInput} from '../pages/Login';
import Adapter from 'enzyme-adapter-react-16';
import renderer from 'react-test-renderer';

configure({ adapter: new Adapter() });



describe('Render Login', () => {

    test('validate email fuction should pass on correct input', () => {
        const email = 'test@test.com';
        expect(validateInput(email)).toBe(true);
    });

    test('validate email fuction should fail on incorrect input', () => {
        const email = 'test';
        expect(validateInput(email)).not.toBe(true);
    });

    test('Hide unhide errors', () => {
        const LoginComponent = render(<Login/>);
        const emailInput = LoginComponent.getByPlaceholderText('Email');
        expect(emailInput.value).toMatch('');
        fireEvent.change(emailInput, {target: {value: 'testing'}});
        expect(emailInput.value).toMatch('testing');

        const errorMsg = LoginComponent.getByText('Invalid email !')
        expect(errorMsg).toBeInTheDocument;
        
        fireEvent.change(emailInput, {target: {value: 'test@test.com'}});
        expect(errorMsg).not.toBeInTheDocument;
      })

    test('validate Log in option is there', () => {
        const LoginComponent = render(<Login/>)
        // console.log(LoginComponent)
        const text = LoginComponent.getByText('Log In')
        expect(text).toBeInTheDocument;
    });

    test('validate Sign Up option is there', () => {
        const LoginComponent = render(<Login/>)
        // console.log(LoginComponent)
        const text = LoginComponent.getByText('Sign Up')
        expect(text).toBeInTheDocument;
    });

    test('validate Email placeholder is there', () => {
        const LoginComponent = render(<Login/>)
        // console.log(LoginComponent)
        const text = LoginComponent.getByPlaceholderText('Email')
        expect(text).toBeInTheDocument;
    });

    test('validate Email placeholder is there', () => {
        const LoginComponent = render(<Login/>)
        // console.log(LoginComponent)
        const text = LoginComponent.getByPlaceholderText('Email')
        expect(text).toBeInTheDocument;
    });

    test('email input uses text type', () => {
        const LoginComponent = render(<Login/>)
        const emailInput = LoginComponent.getByPlaceholderText('Email')
        expect(emailInput.value).toMatch('')
        fireEvent.change(emailInput, {target: {value: 'test@test.com'}})
        expect(emailInput.value).toMatch('test@test.com')
      })

    test('validate Password placeholder is there', () => {
        const LoginComponent = render(<Login/>)
        // console.log(LoginComponent)
        const text = LoginComponent.getByPlaceholderText('Password')
        expect(text).toBeInTheDocument;
    });

    test('password input uses password type', () => {
        const LoginComponent = render(<Login/>)
        const passwordInput = LoginComponent.getByPlaceholderText('Password')
        expect(passwordInput.value).toMatch('')
        fireEvent.change(passwordInput, {target: {value: '****'}})
        expect(passwordInput.value).toMatch('****')
      })

      test('render with no props', () => {  
        const LoginComponent = renderer.create(<Login />).toJSON();
        expect(LoginComponent).toMatchSnapshot();
      });
  });
