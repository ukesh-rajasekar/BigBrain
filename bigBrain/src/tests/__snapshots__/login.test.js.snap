// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render Login render with no props 1`] = `
<div
  className="container-md"
>
  <div
    className="justify-content-md-center row"
  >
    <div
      className="col"
    >
      <div
        className=" text-center card bg-light"
      >
        <div
          className="container card-body"
        >
          <div
            className="card-header"
          >
            Hello BigBrains, please log in
          </div>
          <input
            className="email form-control"
            name="email"
            onChange={[Function]}
            placeholder="Email"
            type="text"
          />
          <input
            className="password form-control"
            name="password"
            onChange={[Function]}
            placeholder="Password"
            type="password"
          />
          <button
            className="btn btn-primary btn-sm"
            disabled={false}
            name="login"
            onClick={[Function]}
            type="button"
          >
            Log In
              
          </button>
          <button
            className="btn btn-primary btn-sm"
            disabled={false}
            name="signup"
            onClick={[Function]}
            type="button"
          >
            Sign Up
              
          </button>
        </div>
      </div>
    </div>
  </div>
</div>
`;
